src/ImageSharp/Formats/Ani/AniHeader.cs

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) Six Labors.
+// Licensed under the Six Labors Split License.
+
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+namespace SixLabors.ImageSharp.Formats.Ani;
+
+internal readonly struct AniHeader
+{
+    public uint Size { get; }
+
+    public uint Frames { get; }
+
+    public uint Steps { get; }
+
+    public uint Width { get; }
+
+    public uint Height { get; }
+
+    public uint BitCount { get; }
+
+    public uint Planes { get; }
+
+    public uint DisplayRate { get; }
+
+    public AniHeaderFlags Flags { get; }
+
+    public static ref AniHeader Parse(ReadOnlySpan<byte> data) => ref Unsafe.As<byte, AniHeader>(ref MemoryMarshal.GetReference(data));
+
+    public void WriteTo(in Stream stream) => stream.Write(MemoryMarshal.AsBytes(MemoryMarshal.CreateReadOnlySpan(in this, 1)));
```

Comment:
```suggestion
    public void WriteTo(Stream stream) => stream.Write(MemoryMarshal.AsBytes(MemoryMarshal.CreateReadOnlySpan(in this, 1)));
```
`Stream` is a reference type, so `in` doesn't provide much value here. Actually it's a [de-optimization](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIAYACY8gOgEkB5AbhvqdYBKAVwB2GAJb4YLAMIR8AB3EAbGFADKagG7iwMXD2q8AzE1IMZDGgG8aDe01PEUDAGIQIACnUZY2fAy4AJQMALwAfAwAqrgwdJ7Bhg6OTC4AQthQ3r4w/oEhEdGx5AlBSQ52DgDaALIwGAAWEAAmbIrKnnWNLW0KyhwKEhAiuCwAchBsIsriIrMA5kEAupX2ClDiWtgYMPxIqUVx2X4BwQzWAL6rVtTJtfVNre2dDz3tA0Mj45PTswvL13Wm22u2Y+2chxKswYPhO+XOV2oFyAA===) as the argument can't be passed in a cpu-register, instead the address needs to be taken,